### PR TITLE
Update CI setup and GitHub organization references

### PR DESCRIPTION
**Pull request by ararslan**
The CI update (first commit) encompasses the following:
- CompatHelper and TagBot need particular permissions set; this commit sets them per the respective repositories' README files.
- Unit testing and documentation deployment are consolidated into one file, in keeping with most other packages in the ecosystem.
- Unit tests are run on:
    - The latest Ubuntu version available on Actions instead of 20.04
    - Julia 1.6, latest 1.x (1.10 as of this writing), and nightly
    - x86_64 (this was always the case but it's now explicit)

Note that I did not update the Codecov action to its most recent version, v4, as that requires a `CODECOV_TOKEN` repository secret. See the Codecov documentation for more information. In the meantime, v2 should continue to work (I think) until it's EOL.

The other two commits are replacing references to the Juice-jl GitHub organization with Tractables.

### CODE REVIEW

File: .github/workflows/tagbot.yml
Comment (khosravipasha) [inline .github/workflows/tagbot.yml:9]: does this mean it looks backs 3 commits and tries to tag all of them (if not already)?

File: .github/workflows/ci.yml
Comment (khosravipasha) [inline .github/workflows/ci.yml:51]: In the past we wanted to trigger doc only builds wihtout running tests but I guess merging them should be fine.

File: .github/workflows/ci.yml
Comment (khosravipasha) [inline .github/workflows/ci.yml]: Does it make sense to keep `workflow_dispatch`? In the past was useful for debuggin the actions.

**Review by khosravipasha — COMMENTED** [review]
Thanks a lot for the changes, mostly LGTM.

Few minor comments inline.

Main questionbefore merging:

1. How can we test the CI run before merging it? To see whether the doc build and upload happend correctly for example.

**Review by khosravipasha — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (ararslan) [reply .github/workflows/ci.yml:51]: Up to you, I can split them back apart if that's a more convenient workflow for you.

File: .github/workflows/tagbot.yml
Comment (ararslan) [reply .github/workflows/tagbot.yml:9]: To be perfectly honest, I'm not sure; I just took that from the [TagBot repository README](https://github.com/JuliaRegistries/TagBot/blob/master/README.md).

**Comment by ararslan** [comment]
> How can we test the CI run before merging it?

Since I'm submitting this PR from a fork, the workflows need to be [approved](https://docs.github.com/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks) before they're run.

File: .github/workflows/ci.yml
Comment (khosravipasha) [reply .github/workflows/ci.yml:51]: Should be okay to merge them.

**Comment by khosravipasha** [comment]
Thanks, seems I don't have access anymore to approve so need to wait until someone else takes a look, and merge this.

**Comment by ararslan** [comment]
Who would still have access? @guyvdbroeck, perhaps?

**Review by guyvdbroeck — APPROVED** [review]

**Comment by ararslan** [comment]
Thanks for the workflow and PR approval, @guyvdbroeck! Looks like CI is happy. I think you'll need to be the one to merge as well given that @khosravipasha doesn't (and obviously I don't) have commit access.

**Comment by ararslan** [comment]
Looking at the CI log now that this is merged, it appears you'll need to regenerate the `DOCUMENTER_KEY` repository secret. While you're at it, it would be worthwhile to also grab the `CODECOV_TOKEN` for this repo from https://codecov.io and add that as a secret. Then the Codecov action can be updated to v4.

**Comment by khosravipasha** [comment]
@ararslan Thanks for checking the logs, seems CI passed but did not deploy because of `DOCUMENTER_KEY `.  I have access again now, I think I know how to do it following these steps: https://documenter.juliadocs.org/stable/man/hosting/#GitHub-Actions

**Comment by ararslan** [comment]
Great! Once you generate a new SSH key and add it as a repository secret, you should be able to rerun the Documentation job triggered by merging this PR, which you can find in the Actions tab.

**Comment by khosravipasha** [comment]
Great Thanks, adding the steps here for future refrences.

1. install DocumenterTools tools

```
(@v1.8) pkg> add DocumenterTools
```

2. Install DirectedAcyclicGraphs in dev mode.

```
(@v1.8) pkg> dev DirectedAcyclicGraphs
```

3. Generate public and private key for documenter_key...

```
Using DocumenterTools
Using DirectedAcyclicGraphs
DocumenterTools.genkeys(DirectedAcyclicGraphs)
```

4. Example Key Generated, and copy in the correct place as instructed in the "info" sections below:

```
julia> DocumenterTools.genkeys(DirectedAcyclicGraphs)
┌ Info: Add the key below as a new 'Deploy key' on GitHub (https://github.com/Tractables/DirectedAcyclicGraphs.jl/settings/keys) with read and write access.
└ The 'Title' field can be left empty as GitHub can infer it from the key comment.

<ssh-rsa KEY....= Documenter> 

[ Info: Add a secure 'Repository secret' named 'DOCUMENTER_KEY' (to https://github.com/Tractables/DirectedAcyclicGraphs.jl/settings/secrets if you deploy using GitHub Actions) with value:

<DOCUMENTER_KEY in base64 encoding.....==>
```

5. Rerun All actions here: https://github.com/Tractables/DirectedAcyclicGraphs.jl/pull/5/checks

**Pull request merged**
